readthedocs/search/tasks.py

```diff
@@ -169,33 +169,30 @@ def record_search_query(project_slug, version_slug, query, total_results, time_s
 
     # If a partial query exists,
     # then just update that object.
-    for partial_query in partial_query_qs.iterator():
+    max_queries = 30
```

Comment:
How did you come to this number? What does it mean as a user in the search context? Will it show only 30 results as maximum?

Reply:
It means that we will only search the latest 30 searches that have happened in the last 10 seconds in that project/version. Just a small protection in case we get flooded.

> Check max 30 queries, in case there is a flood of queries.

Reply:
and this doesn't have anything to do with the search results a user gets, this code is to save in our db analytics about searches that a project has